app/controllers/minor/routes.py

```diff
@@ -36,32 +48,67 @@ def getEngagementInformation(username, type, id, term):
 
     return information
 
-@minor_bp.route('/cceMinor/<username>/addCommunityEngagement', methods=['POST'])
-def addCommunityEngagement(username):
+@minor_bp.route('/cceMinor/<username>/modifyCommunityEngagement', methods=['PUT','DELETE'])
+def modifyCommunityEngagement(username):
     """
-        Saving a term participation/activities for sustained community engagement 
+        Saving a term participation/activities for sustained community engagement
     """
-    pass
+    if not g.current_user.isCeltsAdmin:
+        abort(403)
 
-@minor_bp.route('/cceMinor/<username>/removeCommunityEngagement', methods=['POST'])
-def removeCommunityEngagement(username):
-    """
-        Opposite of above
-    """
-    pass
+    action = 'add' if request.method == 'PUT' else 'remove'
+    try: 
+        setCommunityEngagementForUser(action, request.form, g.current_user)
+    except DoesNotExist:
+        return "There are already 4 Sustained Community Engagement records." 
+    
+    return ""
 
-@minor_bp.route('/cceMinor/<username>/requestOtherCommunityEngagement', methods=['GET,POST'])
-def requestOtherCommunityEngagement(username):
+@minor_bp.route('/cceMinor/<username>/requestOtherCommunityEngagement', methods=['GET', 'POST'])
+def requestOtherEngagement(username):
     """
         Load the "request other" form and submit it.
     """
-    pass
+    user = User.get_by_id(username)
+    terms = selectSurroundingTerms(g.current_term)
+    
+
+    if request.method == 'POST':
+        flash("Other community engagement request submitted.", "success")
+        attachmentName = None
+        attachment = request.files.get("attachmentObject")
+        if attachment:
+                addFile= FileHandler(getFilesFromRequest(request))
+                addFile.saveFiles()
+                attachmentName = attachment.filename
+        formData = request.form.copy()
+        formData["attachment"] = attachmentName
+        saveOtherEngagementRequest(formData)
+        return redirect(url_for("minor.viewCceMinor", username=user))
+
+
+    return render_template("/minor/requestOtherEngagement.html",
+                            user=user,
+                            terms=terms)
+
+
+@minor_bp.route("/deleteRequestFile", methods=["POST"])
+def deleteRequestFile():
+
+    fileData= request.form
+    termFile=FileHandler(termId=fileData["databaseId"])
+    termFile.deleteFile(fileData["fileId"])
+
+    return ""
 
 @minor_bp.route('/cceMinor/<username>/addSummerExperience', methods=['POST'])
 def addSummerExperience(username):
-    pass
+    saveSummerExperience(username ,request.form, g.current_user)

```

Comment:
I am a stickler so remove the space in between the variable and the comma hee hee.